.github/workflows/check-cabal-files.yml

```diff
@@ -0,0 +1,20 @@
+name: Haskell Cabal Check
+
+on:
+  push:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Install Haskell
+      uses: input-output-hk/setup-haskell@v1
+      id: setup-haskell
+      with:
+        cabal-version: "3.8.1.0"
+
+    - uses: actions/checkout@v3
+
+    - name: Cabal check
+      run: ./scripts/check-cabal-files.sh
```

Comment:
Can you move this to `build.yml` as a separate job? Advantages:

 - The `on:` triggers are reused (e.g. they are incorrect here, as they are not triggered on PRs).
 - We don't have to update the branch protection rules (see the "Required" tags) as well as `bors.toml` (once we start using it again).

Reply:
Moved!